source/test/TestApp/PlayFabEventTest.cpp

```diff
@@ -369,9 +362,11 @@ namespace PlayFabUnit
         return api;
     }
 
-    std::unique_ptr<PlayFab::PlayFabEvent> PlayFabEventTest::MakeEvent(PlayFab::PlayFabEventType eventType)
+    std::unique_ptr<PlayFab::PlayFabEvent> PlayFabEventTest::MakeEvent(TestContext& testContext, PlayFab::PlayFabEventType eventType)
     {
-        auto event = std::unique_ptr<PlayFab::PlayFabEvent>(new PlayFab::PlayFabEvent());
+        auto testEvent = new TestEvent();
+        testEvent->testContext = &testContext;
+        auto event = std::unique_ptr<PlayFab::PlayFabEvent>(testEvent);
```

Comment:
Simplify this by allocating TestEvent directly into the unique_ptr and then dereference the unique_ptr to set testContext.

Reply:
Your suggestion feels like a style issue, and de-ref-ing and re-casting the unique ptr seems a lot uglier than just keeping the ref to the original pointer.